internal/dag/dag.go

```diff
@@ -79,10 +80,16 @@ func (d *DAG) Visit(fn func(Vertex)) {
 	}
 }
 
-// Statuses returns a slice of Status objects associated with
-// the computation of this DAG.
-func (d *DAG) Statuses() map[types.NamespacedName]Status {
-	return d.statuses
+// GetTestStatuses returns a slice of Status objects associated with
+// the computation of this DAG, for testing status output.
```

Comment:
Is the function defined to only emit HTTPProxy conditions? Can we add a TODO and an issue to remove it?

Reply:
I've renamed it. We won't be able to remove it completely without completely refactoring the status tests, as they need the name of the object and the detailed object. That will need a lot of extra testing machinery to sit between the `status.Cache` and the status tests.

#2967 opened to track.